### PR TITLE
8.2 questionnaire

### PR DESCRIPTION
**Pull request by lyrated**
questionnaire changed to multistep form, added data validation

### CODE REVIEW

**Review by joshuameiser — APPROVED** [review]
I've pulled your changes and tried the signup for both recruiter and candidate. Everything worked flawlessly. Great work!
It's an okay from me 👍

File: public/javascripts/questionnaires/questionnaire.js
Comment (simbuzan) [inline public/javascripts/questionnaires/questionnaire.js:44]: The function is so long I think it would be better to use more specific names instead of letters. It would make the code more readable.
You could of course leave the iterator names as they are (i,j).

**Comment by simbuzan** [comment]
I tired out the changes and it is now way more intuitive! 
The only thing I noticed was that the error messages is still shown when I come back to edit my answer, which might make it a bit confusing, but I don't think it is a necessary change, since the user is not constricted anymore. 
The form worked for both developer and employer.
🕐 Waiting for answer on code comment, otherwise approved.

**Comment by lyrated** [comment]
Thanks for the reviews!
@simbuzan changed the variables, but only in that one function :P since we copied the code, we just left it as it is

Will merge after @fwoodmann is done with his part~

File: public/javascripts/questionnaires/questionnaire.js
Comment (mbareeva) [inline public/javascripts/questionnaires/questionnaire.js:23]: where do you call this function? nextPrev(n)? I see you call on showTab

**Comment by mbareeva** [comment]
great work with tabs!

File: public/javascripts/questionnaires/questionnaire.js
Comment (lyrated) [reply public/javascripts/questionnaires/questionnaire.js:23]: this is the code we copied and didn't write btw :P but this function is called when you click on a button (see the ejs files where the buttons are)

**Pull request merged**
